src/sync/sync_session.cpp

```diff
@@ -427,13 +427,19 @@ void SyncSession::handle_progress_update(uint64_t downloaded, uint64_t downloada
         m_current_downloadable = downloadable;
         m_current_downloaded = downloaded;
         m_current_uploaded = uploaded;
+        std::vector<uint64_t> to_erase;
         for (auto& pair : m_notifiers) {
             bool should_delete = false;
-            invocations.emplace_back(create_notifier_invocation(pair.second, should_delete));
+            auto package = pair.second;
+            invocations.emplace_back(create_notifier_invocation(package, should_delete));
             if (should_delete) {
-                m_notifiers.erase(pair.first);
+                to_erase.emplace_back(pair.first);
             }
         }
+        // We need to erase after iteration since our Linux builds break if we try it the C++14 way.
```

Comment:
I'll note GCC 5.4.0 at least, since that was the version on my VM where I reproduced the issue.